python/ray/experimental/client/dataclient.py

```diff
@@ -0,0 +1,104 @@
+"""
+This file implements a threaded stream controller to abstract a data stream
+back to the ray clientserver.
+"""
+import logging
+import queue
+import threading
+
+from typing import Any
+from typing import Dict
+
+import ray.core.generated.ray_client_pb2 as ray_client_pb2
+import ray.core.generated.ray_client_pb2_grpc as ray_client_pb2_grpc
+
+logger = logging.getLogger(__name__)
+
+# The maximum field value for request_id -- which is also the maximum
+# number of simultaneous in-flight requests.
+INT32_MAX = (2**31) - 1
+
+
+class DataClient:
+    def __init__(self, channel, client_id):
```

Comment:
type annotation.

Reply:
`channel` is a judgement call here because it's part of a private module to grpc -- I've added a string annotation but I don't like the idea of importing from some other libraries internals.